src/App.js

```diff
@@ -1,23 +1,51 @@
-import logo from './logo.svg';
-import './App.css';
+import ForecastTable from "./components/ForecastTable";
+import CurrentLocationForecastBtn from "./components/CurrentLocationForecastBtn";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchForecast } from "./actions";
+import { useState } from "react";
 
 function App() {
+  const [cityName, setCityName] = useState("");
+  const dispatch = useDispatch();
+  const defaultCity = useSelector((state) => state.defaultCity);
+
+  if (defaultCity) {
```

Comment:
Ideally, I think you'd want to use `useEffect` here:

```js
useEffect(function () {
    if (defaultCity) {
      console.log('called')
      dispatch(fetchForecast(defaultCity));
    }
  }, [defaultCity])
```

Often times you may not really know how many times a component will be rendered, so if this component re-rendered many times, then this request (the one that is invoked as a result of this dispatch) would happen many times, which isn't great.

Moreover, you risk creating a looping re-render - thankfully not in this case. However, if this request caused a prop on this component to update, and then caused the component to re-render, it'd happen over and over again.

Not only does `useEffect` help with the timing of our request (after the component renders) but using the dependency array, we prevent the request from happening again if the value (in this case, `defaultCity`), hasn't changed.